### PR TITLE
Add Name() to export resource interface

### PR DESCRIPTION
**Pull request by calufornia**
## Description

There are parts of the local dev server that require the resource's name (we've been using the resource slug instead) - this adds a method to the export resource interface which returns the resource's name, similar to how `ID()` returns the resource's ID.

Also deletes some unused resource functions.

## Test plan
n/a

### CODE REVIEW

**Review by francescaleung — COMMENTED** [review]
wait, this is dumb. Since everything already embeds BaseResource (& should), can we just have getters on the Resource interface & put those getters on the BaseResource?

i.e.
```
func (r BaseResource) GetName() string {
  return r.Name
}
```

& then you don't need to individually define it for everything else?

Sometimes I forget how struct composition works in Go.

**Comment by calufornia** [comment]
> wait, this is dumb. Since everything already embeds BaseResource (& should), can we just have getters on the Resource interface & put those getters on the BaseResource?
> 
> i.e.
> 
> ```
> func (r BaseResource) GetName() string {
>   return r.Name
> }
> ```
> 
> & then you don't need to individually define it for everything else?
> 
> Sometimes I forget how struct composition works in Go.

Better yet, I think we can define a `BaseResource` interface, and then embed that interface in the `Resource` interface (not sure if that's what you meant by having the getters on the `Resource` interface), I'll update the PR

File: pkg/resources/resource.go [hunk 1]
```diff
@@ -2,21 +2,40 @@ package resources
...
 
-	"github.com/mitchellh/mapstructure"
 	"github.com/pkg/errors"
 )
 
+type IBaseResource interface {
+	GetKind() ResourceKind
+	GetID() string
+	GetSlug() string
+	GetName() string
```
Comment (francescaleung) [inline pkg/resources/resource.go:13]: can we deprecate Kind() & ID() on the Resource interface?

**Review by francescaleung — APPROVED** [review]
assuming the removed functions are unused, lgtm

File: pkg/resources/resource.go [hunk 2]
```diff
@@ -36,56 +55,6 @@ type ResourceKind string
...
-	return GetAirplaneEnvFromFunc(ref, name, os.LookupEnv)
-}
-
-func GetAirplaneEnvFromFunc(ref string, name string, f EnvLookupFunc) string {
-	key := fmt.Sprintf("AIRPLANE_%s_%s", strings.ToUpper(ref), strings.ToUpper(name))
-	val, _ := f(key)
-	return val
-}
-
-func AirplaneResourceFromFunc(ref string, f EnvLookupFunc, res Resource) error {
```
[the remark this answers is not in the document]
Comment (calufornia) [reply pkg/resources/resource.go:49]: yeah! only `GetAirplaneResourceFromFunc` is used

File: pkg/resources/resource.go (hunk 1 above)
Comment (calufornia) [reply pkg/resources/resource.go:13]: yep, we should change any callers of `Kind()` and `ID()` to use `GetKind()` and `GetID()` before we remove them though. Ideally the base resource interface methods would just be `Kind()`, `ID()`, etc. but before we can do that we'll need to make the Base Resource methods private (or change their names) so as not to have the field and function names conflict

Comment (calufornia) [reply pkg/resources/resource.go:13]: I'll mark them as deprecated.

**Pull request merged**